libevmasm/SemanticInformation.cpp

```diff
@@ -262,13 +262,16 @@ bool SemanticInformation::invalidatesMemory(Instruction _instruction)
 	case Instruction::CALLCODE:
 	case Instruction::DELEGATECALL:
 	case Instruction::STATICCALL:
-		return true;
+		return yul::SideEffects::Write;
+	case Instruction::KECCAK256:
+	case Instruction::MLOAD:
```

Comment:
We could include `msize` here, just to be sure.

Reply:
Should also add the following:
 - return
 - revert
 - log0 - log4

Reply:
Also STOP, I think.  Where about INVALID?

Reply:
neither stop nor invalid read or write to memory. They should be marked as non-movable and having side-effects though.